src/Stratis.Bitcoin.Features.Consensus/Interfaces/ISmartContractCoinViewRuleLogic.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Threading.Tasks;
+using NBitcoin;
+using Stratis.Bitcoin.Base.Deployments;
+using Stratis.Bitcoin.Consensus.Rules;
+
+namespace Stratis.Bitcoin.Features.Consensus.Interfaces
+{
+    public interface ISmartContractCoinViewRuleLogic
+    {
+        Task RunAsync(Func<RuleContext, Task> baseRunAsync, RuleContext context);
+
+        /// <summary>
+        /// Executes contracts as necessary and updates the coinview / UTXOset after execution.
+        /// </summary>
+        void UpdateCoinView(Action<RuleContext, Transaction> baseUpdateUTXOSet, RuleContext context, Transaction transaction);
+
+        bool CheckInput(Func<Transaction, int, TxOut, PrecomputedTransactionData, TxIn, DeploymentFlags, bool> baseCheckInput, Transaction tx, int inputIndexCopy, TxOut txout, PrecomputedTransactionData txData, TxIn input, DeploymentFlags flags);
```

Comment:
Please add comment, everything inside Stratis.Bitcoin.Features.Consensus should be well documented

Reply:
Unfortunately there were no comments to move over from the original class when I added the interface. It would have been useful if the rule you are suggesting had been followed up to now. I will try to figure out some comments or we can leave it to the original author to add those..

Reply:
Done